app/validators/links_validator.rb

```diff
@@ -0,0 +1,19 @@
+class LinksValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, links)
+    invalidly_formatted = invalid_formatted_links(links)
+    if invalidly_formatted.any?
+      record.errors.add(attribute, "#{invalidly_formatted.to_sentence} has a value with an invalid format.")
+    end
+  end
+
+private
+
+  def invalid_formatted_links(links)
+    invalid = links.select do |_key, link_values|
+      link_values.flat_map(&:last)
+        .any? { |link| !link.to_s.match?(/\A[a-zA-Z0-9\-_\/]*\z/) }
```

Comment:
Links should be UUID values so we can probably use the regex from UuidValidator: https://github.com/alphagov/email-alert-api/blob/master/app/validators/uuid_validator.rb#L2